### DISCUSSION TITLE
"Just-pair" seems to not work in the peripheral-HID sample

### DISCUSSION THREAD

**Discussion by malto101** [zephyrproject-rtos/zephyr#78131] [Q&A] [answered]
I have a nrf52840dongle with me, upon disabling the Kconfig Setting CONFIG_SAMPLE_BT_USE_AUTHENTICATION, the Just-pair setting seems to not work when trying to connect to a phone, but when we enable the CONFIG_SAMPLE_BT_USE_AUTHENTICATION KConfig, it works as intended.

The logs are below

```
*** Booting Zephyr OS build v3.7.0-2211-g4677920956ed ***
[00:00:00.330,627] <inf> fs_nvs: 4 Sectors of 4096 bytes
[00:00:00.330,627] <inf> fs_nvs: alloc wra: 1, c38
[00:00:00.330,657] <inf> fs_nvs: data wra: 1, 804
[00:00:00.332,946] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
[00:00:00.332,977] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)
[00:00:00.333,007] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 3.7 Build 99
[00:00:00.333,282] <inf> bt_hci_core: No ID address. App must call settings_load()
Bluetooth initialized
[00:00:00.337,341] <inf> bt_hci_core: Identity: F2:63:3C:72:BD:86 (random)
[00:00:00.337,371] <inf> bt_hci_core: HCI: version 5.4 (0x0d) revision 0x0000, manufacturer 0x05f1
[00:00:00.337,402] <inf> bt_hci_core: LMP: version 5.4 (0x0d) subver 0xffff
Advertising successfully started
Connecting to 94:8A:C6:7D:A4:76 (public)
Security failed: 94:8A:C6:7D:A4:76 (public) level 1 err 4
Disconnected from 94:8A:C6:7D:A4:76 (public) (reason 0x13)
```

[wireshark.log](https://github.com/user-attachments/files/16922438/wireshark.log)
I have attached the wireshark logs from my blesniffer above

Regards
Dhruv Menon

**Comment by koffes** [comment]
Why do you want to disable this config?

At least from the logs, the phone is the one who terminates the connection (reason 0x13 = Remote user terminated connection). So it seems the phone mandates a level of security and the kit is unable to comply with that setting.

**Reply by malto101** [reply]
In the HID peripheral sample present in Zephyr,
There is a custom KConfig option called SAMPLE_BT_USE_AUTHENTICATION which enable/disables passkey authentication, If disabled it defaults to connecting without authentication.

I see, Could you suggest any particular changes to be done, And Around 2023, when I first dabbled in zephyr, I used to be able connect to my mobilePhone seamlessly using the same sample.

And as the sample is a user(phone) initiated connection, I suppose it shouldn't disconnect from the phone end(correct me if I am wrong)

**Reply by koffes** [reply, marked as answer]
I am not that ventured in this sample.
One tip is to check in the `.config` file that CONFIG_BT_SMP=y.
I would try with Another Phone or PC to see what difference that makes.

It could be that the phone has received a software update, and now mandates a (higher) level of security for HID. The "err 4" is 
`/** The requested security level could not be reached. */
	BT_SECURITY_ERR_AUTH_REQUIREMENT,`
	
So what happens is that after connection, the phone requests a security update. The HID is unable to comply so the phone terminates the connection.

**Reply by malto101** [reply]
Hello @koffes ,
Thank you for your prompt reply,
I just check with another phone, and I was able to discover a new piece.

It was able to successfully connect to another device.
But if I forget the connection and retry on the same device. It doesn't allow the connection to continue(Thus terminating it)

After I was unable to successfully connect to the 2nd device, I retried it the orginal phone and was able connect it flawlessly.
but after forgetting the connection, it follows the same pattern as above.

Do you have any inkling to why does it happen, and how to mitigate this.

**Reply by malto101** [reply]
enabling debug showed the below error when we connect to new device
`[00:05:37.093,841] <err> bt_gatt: unable to save SC: no cfg left
`

**Reply by koffes** [reply]
If I understand correctly, you connect the HID device to phone A, forget the connection on the phone and then it will not connect again.
What may happen here is that your phone has an address XX. When you forget the connection it clears the phone key, when you try to connect again the HID device is already bonded to the phone with address XX but the phone keys are gone, and the connection is terminated.

To test, try to to see what happens if you turn off Bluetooth all together on the phone and/or restart it. That may change the address of the phone to YY, the HID device sees this as a new device and a new set of keys are generated.

**Reply by malto101** [reply]
Just tested it.
Sadly it was unsuccessful. the addresses didn't change
I second your reasoning to why it isn't connecting

hmm,
Maybe if there is way erase all set of the keys, this could be less of a hassle,
I have already enabled `CONFIG_BT_KEYS_OVERWRITE_OLDEST=y`
but it would be better if it could remove the all of them
